From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Tue, 22 Jul 2025 22:31:58 +0200
Subject: [PATCH] Op lock


diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index e2f00f7d5328dac16fffaafef021c2cfc569c9a5..a5dcd47c75a0b3d18a6b224eec8d96ca8e69459c 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -1200,6 +1200,9 @@ public abstract class PlayerList {
     }
 
     public void op(GameProfile profile) {
+        if (org.dreeam.leaf.config.modules.misc.OpLock.preventOpChanges) {
+            return;
+        }
         this.ops.add(new ServerOpListEntry(profile, this.server.getOperatorUserPermissionLevel(), this.ops.canBypassPlayerLimit(profile)));
         ServerPlayer player = this.getPlayer(profile.getId());
         if (player != null) {
@@ -1208,6 +1211,9 @@ public abstract class PlayerList {
     }
 
     public void deop(GameProfile profile) {
+        if (org.dreeam.leaf.config.modules.misc.OpLock.preventOpChanges) {
+            return;
+        }
         this.ops.remove(profile);
         ServerPlayer player = this.getPlayer(profile.getId());
         if (player != null) {
